Incompatible with PyTorch 2.6

### Describe the bug

When rebuilding for PyTorch 2.6, a compilation error throws. See https://github.com/conda-forge/abacus-feedstock/pull/55.

```
[ 41%] Building CXX object source/module_hamilt_lcao/hamilt_lcaodft/CMakeFiles/hamilt_lcao.dir/LCAO_nl_mu.cpp.o
/home/conda/feedstock_root/build_artifacts/abacus_1739621401458/work/source/module_hamilt_lcao/module_deepks/LCAO_deepks_torch.cpp: In member function 'void LCAO_Deepks::cal_gvdm(int)':
/home/conda/feedstock_root/build_artifacts/abacus_1739621401458/work/source/module_hamilt_lcao/module_deepks/LCAO_deepks_torch.cpp:119:54: error: 'torch::linalg' has not been declared
  119 |             torch::Tensor tmp_y = std::get<0>(torch::linalg::eigh(tmp_x, "U"));
      |                                                      ^~~~~~
```

### Expected behavior

No error.

### To Reproduce

See https://github.com/conda-forge/abacus-feedstock/pull/55.

### Environment

See https://github.com/conda-forge/abacus-feedstock/pull/55.

### Additional Context

In the PyTorch documentation: [at::linalg_eigh](https://pytorch.org/cppdocs/api/function_namespaceat_1a7f73ff1b9df8c41d4572de164e0a4761.html#_CPPv4N2at11linalg_eighERKN2at6TensorEN3c1011string_viewE)

### Task list for Issue attackers (only for developers)

- [ ] Verify the issue is not a duplicate.
- [ ] Describe the bug.
- [ ] Steps to reproduce.
- [ ] Expected behavior.
- [ ] Error message.
- [ ] Environment details.
- [ ] Additional context.
- [ ] Assign a priority level (low, medium, high, urgent).
- [ ] Assign the issue to a team member.
- [ ] Label the issue with relevant tags.
- [ ] Identify possible related issues.
- [ ] Create a unit test or automated test to reproduce the bug (if applicable).
- [ ] Fix the bug.
- [ ] Test the fix.
- [ ] Update documentation (if necessary).
- [ ] Close the issue and inform the reporter (if applicable).

--- a/source/module_hamilt_lcao/module_deepks/LCAO_deepks_torch.cpp
+++ b/source/module_hamilt_lcao/module_deepks/LCAO_deepks_torch.cpp
@@ -115,7 +115,7 @@ void LCAO_Deepks::cal_gevdm(const int nat, std::vector<torch::Tensor>& gevdm)
             // repeat each block for nm times in an additional dimension
             torch::Tensor tmp_x = this->pdm[inl].reshape({nm, nm}).unsqueeze(0).repeat({nm, 1, 1});
             // torch::Tensor tmp_y = std::get<0>(torch::symeig(tmp_x, true));
-            torch::Tensor tmp_y = std::get<0>(torch::linalg::eigh(tmp_x, "U"));
+            torch::Tensor tmp_y = std::get<0>(torch::linalg_eigh(tmp_x, "U"));
             torch::Tensor tmp_yshell = torch::eye(nm, torch::TensorOptions().dtype(torch::kFloat64));
             std::vector<torch::Tensor> tmp_rpt; // repeated-pdm-tensor (x)
             std::vector<torch::Tensor> tmp_rdt; // repeated-d-tensor (y)

--- a/source/module_hamilt_lcao/module_deepks/cal_descriptor.cpp
+++ b/source/module_hamilt_lcao/module_deepks/cal_descriptor.cpp
@@ -67,7 +67,7 @@ void LCAO_Deepks::cal_descriptor(const int nat)
         std::tuple<torch::Tensor, torch::Tensor> d_v(this->d_tensor[inl], vd);
         // d_v = torch::symeig(pdm[inl], /*eigenvalues=*/true,
         // /*upper=*/true);
-        d_v = torch::linalg::eigh(pdm[inl], /*uplo*/ "U");
+        d_v = torch::linalg_eigh(pdm[inl], /*uplo*/ "U");
         d_tensor[inl] = std::get<0>(d_v);
     }
     ModuleBase::timer::tick("LCAO_Deepks", "cal_descriptor");

